src/server/replica.cc

```diff
@@ -303,10 +303,17 @@ std::error_code Replica::HandleCapaDflyResp() {
   }
 
   // If we're syncing a different replication ID, drop the saved LSNs.
-  if (master_context_.master_repl_id != ToSV(LastResponseArgs()[0].GetBuf())) {
+  string_view master_repl_id = ToSV(LastResponseArgs()[0].GetBuf());
```

Comment:
"This behavior change on replica side should be under flag with default to preserve the current behaviour."

Reply:
I completely missed that :facepalm: